### ISSUE TITLE
jws.VerifySet() should return the verification key as well

### ISSUE THREAD

**Issue by anatol** [lestrrat-go/jwx#385]
jws.VerifySet() accepts a signed message plus a keyset and returns the verified message.

It would be great to return the key that been used to sign the message as well. Such key might be later processed, e.g. in case of `clevis` there is further key thumbprint checking.

**Comment by lestrrat** [comment]
Huh, didn't think of that. I guess the easiest way would be introduce an option?

```go
var keyUsed jwk.Key
_, _ = jws.VerifySet(data, keyset, jws.WithKeyUsed(&keyUsed))

// the key used would be sured in keyUsed now
```

**Comment by anatol** [comment]
Yeah, change the `jws.VerifySet` function signature like this.

I am not sure about adding a new option though. Is `jws.WithKeyUsed(&keyUsed))` really needed?

**Comment by lestrrat** [comment]
Well, if you want the use this feature _now_, I would use that `jws.WithKeyUsed` option. I don't want to release a new major version just for this change.

**Comment by lestrrat** [comment]
7cffd9f would be my proposed solution for the time being

**Comment by anatol** [comment]
I can wait for a proper solution. For now I can just copy `VerifySet` to my project and add functionality required.

**Comment by lestrrat** [comment]
Ah, okay. In that case, I will take note of this, create a branch or something and keep this new change there until we can merge and release

**Comment by lestrrat** [comment]
@anatol Here you go: 4ae001a
If this is looks okay, I'll probably create a  branch like `develop/next-major-release` or something and keep the changes there for the time being.

**Comment by anatol** [comment]
Thanks. Your implementation looks good to me.

**Comment by lestrrat** [comment]
This is implemented, and being tracked in #388. Closing for now.
Thanks!
